src/map/map.cpp

```diff
@@ -345,6 +345,31 @@ TIMER_FUNC(map_freeblock_timer){
 	return 0;
 }
 
+FreeBlockLock::FreeBlockLock(bool startLocked) {
+	if (startLocked) {
+		lock();
+	}
+}
+
+FreeBlockLock::~FreeBlockLock() {
+	unlock();
+}
+
+void FreeBlockLock::lock() {
+	if (!locked) {
+		map_freeblock_lock();
+		locked = true;
+	}
+}
+
+void FreeBlockLock::unlock() {
+	if (locked) {
+		map_freeblock_unlock();
+		locked = false;
+	}
```

Comment:
Shouldnt we warn/error if it was not even locked?

Reply:
I don't think so, because we could choose to unlock it manually, and then when it's destroyed it'll try to unlock again, which is fine.

Reply:
Shouldnt the destructor check the "locked" status and only call unlock, when necessary?